devicetypes/automateasia/wyfy-touch-zwave-switch.src/wyfy-touch-zwave-switch.groovy

```diff
@@ -0,0 +1,522 @@
+/**
+ *  WYFY TOUCH ZWAVE SWITCH
+ *
+ *  Copyright 2020 Edwin Tan Poh Heng
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+
+/*
+Description of State variables
+endpointCount: The number of endpoints
+endpointMap: A collection that maps childDeviceNetworkId to their endpoint 
+zwaveParemeterKey: A collection of the ZWave parameter's value and state
+*/
+
+metadata {
+	definition (name: "WYFY TOUCH ZWAVE SWITCH", namespace: "AutomateAsia", author: "Edwin Tan Poh Heng", cstHandler: true) {
+		capability "Zw Multichannel"
+		capability "Switch"
+		capability "Refresh"
+		capability "Configuration"
+		capability "Health Check"
+		
+		// Batch 1
+		fingerprint mfr: "015F", prod: "3102", model: "0201", deviceJoinName: "WYFY Touch S1"
+		fingerprint mfr: "015F", prod: "3102", model: "0202", deviceJoinName: "WYFY Touch S1"
+		fingerprint mfr: "015F", prod: "3102", model: "0204", deviceJoinName: "WYFY Touch S1"
+		
+		// Batch 2 (model 5102)
+		fingerprint mfr: "015F", prod: "3111", model: "5102", deviceJoinName: "WYFY Touch S1" // 1-button variant
```

Comment:
Please change _deviceJoinName_ to `deviceJoinName: "WYFY Switch 1"`

Reply:
Done. I have change the deviceJoinName to deviceJoinName: "WYFY Touch 1" since the product is called WYFY Touch.

Reply:
Still, please rename it to `"WYFY Switch 1"`
We've decided to keep devices' names across whole system more unified, so they should follow `<manufacturer> + <device type>`

Reply:
Have renamed to "WYFY Switch 1". See https://github.com/erocm123/SmartThingsPublic/pull/23